packages/api/cms-api/src/mail-templates/mail-template.service.ts

```diff
@@ -0,0 +1,98 @@
+import { DiscoveryService } from "@golevelup/nestjs-discovery";
+import { EntityManager } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { Options as MailOptions } from "nodemailer/lib/mailer";
+import { createIntl, createIntlCache, IntlCache, IntlShape } from "react-intl";
+import { ContentScope } from "src/user-permissions/interfaces/content-scope.interface";
+
+import { MailerService } from "../mailer/mailer.service";
+import { MAIL_TEMPLATE_METADATA_KEY, MailTemplateMetadata } from "./mail-template.decorator";
+
+export type MailTemplateInterface<T> = {
+    id: string;
```

Comment:
If we change to a service-based approach, we can probably remove the `id` param as well and use the injection token.